### DISCUSSION TITLE
Internal API Call in the same NextJS App not working - 404 not found

### DISCUSSION THREAD

**Discussion by ZhaniWalid** [vercel/next.js#57924] [Help]
### Summary

Hello I followed this tutorial, to create an internal API EndPoint as server and make the API call in React with 'use client'. But it didn't worked for me: [https://nextjs.org/learn-pages-router/basics/api-routes/creating-api-routes](https://nextjs.org/learn-pages-router/basics/api-routes/creating-api-routes)

I placed the 'hello.ts' file under 'src/pages/api' and then under 'src/app/pages/api' and make the call with axios like this:

```
const dataAxios = axios.get('../pages/api/hello') // calling the file in the folder 'src/app/pages/api'
    .then(
      res => console.log('dataAxios - res: ', res)
    ).catch(
      err => console.error('dataAxios - error: ', err)
    );

  useEffect(() => {
    console.log('Data images -- dataAxios -- from public static folder:', JSON.stringify(dataAxios));
  }, [dataAxios]);
```

And i got this error: `GET http://localhost:3000/pages/api/get-images-from-public-folder 404 (Not Found)`

`dataAxios - error:  AxiosError {message: 'Request failed with status code 404', name: 'AxiosError', code: 'ERR_BAD_REQUEST', config: {…}, request: XMLHttpRequest, …}`

**Comment by icyJoseph** [comment]
Hi,

You need to get it from `axios.get("/api/hello")`

Note that if during build time this is not going to work so well. And if you are already server side, you better do it directly there.

As I see you have it, you are also not using asynchronous promises correctly. 

For example you should do:

```js
useEffect(() => {
    async function waitForData() {
        const data = await dataAxios;
        console.log(data) 
    }
    waitForData();
})
```

Also your `.then` block is not returning data, so `JSON.stringify(data)` wouldn't do what you think...

**Comment by ZhaniWalid** [comment]
Hello, thank you for your help. Yeah it worked now like this `axios.get("/api/hello")`.

For this point **Note that if during build time this is not going to work so well**, how to deal with it then?

**Reply by icyJoseph** [reply]
I'm a bit busy now, but if you are just learning, I think first you should read into using data fetching in React, without server components. 

What won't necessarily work during build time, is trying to fetch data from API routes as you build. For example within get static props or within server components.

Only in those cases, the solution is to just fetch the data from its source on the same spot. Read more here: https://nextjs-faq.com/fetch-api-in-rsc

**Comment by ZhaniWalid** [comment]
Hello again, thank you for your time.
 
I was used to work with CRA (Create React App) for front-end, and now I am learning NextJS, and i was trying to make that **Internal API Call in the same NextJS App** for days. For that reason I asked here.

Thank you very much for your help anyway, much appreciated.
